tests/unit/browser/test_qutehistory.py

```diff
@@ -0,0 +1,117 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2017 Imran Sobir
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+from PyQt5.QtCore import QUrl
+from qutebrowser.browser import history, qutescheme
+from qutebrowser.utils import objreg
+import datetime
+import pytest
+import tempfile
+
+
+class TestHistoryHandler:
+    """Test the qute://history endpoint."""
+
+    @pytest.fixture(autouse=True)
+    def fake_objects(self, fake_save_manager):
+        """ Create fake web-history with history for three different days """
+        try:
+            original_save_manager = objreg.get('save-manager')
+        except KeyError:
+            original_save_manager = None
+
+        try:
+            original_web_history = objreg.get('web-history')
+        except KeyError:
+            original_web_history = None
```

Comment:
I wouldn't bother with restoring the old objects, it's probably just a test not cleaning up after itself. See #1652. If nothing fails because of that, just drop the restoring and just clean up after the test.